jbpm-flow/src/main/java/org/jbpm/workflow/instance/impl/WorkflowProcessInstanceImpl.java

```diff
@@ -132,11 +136,19 @@ public int getLevelForNode(String uniqueID) {
     }
 
 	public void removeNodeInstance(final NodeInstance nodeInstance) {
+        InternalKnowledgeRuntime kruntime = getKnowledgeRuntime();
+        ProcessEventSupport processEventSupport = ((InternalProcessRuntime) kruntime.getProcessRuntime()).getProcessEventSupport();
+        processEventSupport.fireBeforeNodeRemoved(nodeInstance, kruntime);
+
 		if (((NodeInstanceImpl) nodeInstance).isInversionOfControl()) {
 			getKnowledgeRuntime().delete(
 					getKnowledgeRuntime().getFactHandle(nodeInstance));
 		}
 		this.nodeInstances.remove(nodeInstance);
+		if( isQueueBased() ) {
+		    removeAssociatedAction(nodeInstance);
+		}
+//		System.out.println( "<-" + nodeInstance.getNodeId() + ">");
```

Comment:
remove system out and add fireAfterNodeRemoved from event support (once added) for consistency of event listeners